src/searcher/mod.rs

```diff
@@ -349,43 +351,36 @@ impl Searcher {
         Ok(())
     }
 
+    pub fn search_slice_callback_no_path<'a, 'text, 'tree>(
+        &mut self,
+        query_context: QueryContext,
+        // slice: impl TextProvider<'a> + Parseable + 'a,
+        slice: impl Into<RopeOrSlice<'text>>,
+        tree: Option<&'tree Tree>,
+        mut callback: impl FnMut(&QueryMatch),
+    ) -> Result<(), ConfigError> {
+        self.check_config()?;
+
+        log::trace!("slice reader: searching via multiline strategy");
+        get_matches(query_context.language, slice, &query_context.query, tree).for_each(
```

Comment:
I added a `get_matches()` corresponding to the `get_captures()` "iterate over query captures" abstraction